### PR TITLE
Vulkan-Headers: Use known_good.json for Vulkan-Headers

### PR DESCRIPTION
**Pull request by 1480c1**
Currently testing it.

#1147

### CODE REVIEW

File: build/media-suite_compile.sh [hunk 1]
```diff
@@ -1825,13 +1825,21 @@ if [[ $mpv != "n" ]] && pc_exists libavcodec libavformat libswscale libavfilter;
     fi
 
     if ! mpv_disabled vulkan; then
+        # Get the known_good json and get the commit version of the Vulkan-Headers.
+        # It is assumed that the version number has quotes around them (sed)
+        [[ "$jq" != y ]] && do_pacman_install jq
+        _Vulkan_Version=$(
+            ${curl_opts[@]} https://raw.githubusercontent.com/KhronosGroup/Vulkan-ValidationLayers/master/scripts/known_good.json |
+            jq  '.repos[1].commit' |
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: You can use `-r` to output raw strings.

Comment (1480c1) [reply build/media-suite_compile.sh]: Oh.

File: build/media-suite_compile.sh [hunk 2]
```diff
@@ -1825,13 +1825,21 @@ if [[ $mpv != "n" ]] && pc_exists libavcodec libavformat libswscale libavfilter;
...
+            sed -e 's/^"//;s/"$//')
```
Comment (1480c1) [inline build/media-suite_compile.sh]:
```suggestion
```

File: build/media-suite_compile.sh (hunk 1 above)
Comment (1480c1) [inline build/media-suite_compile.sh]:
```suggestion
            jq -r '.repos[1].commit')
```

Comment (1480c1) [reply build/media-suite_compile.sh]: I wasn't sure what it meant by raw instead of json

**Comment by 1480c1** [comment]
Worked for me

File: build/media-suite_compile.sh (hunk 1 above)
Comment (wiiaboo) [reply build/media-suite_compile.sh]: `jq --raw-output '.repos[] | select(.name == "Vulkan-Headers").commit'` should supposedly work, not apparently not in Windows.

**Comment by wiiaboo** [comment]
Oh, but you're still using array index, not the actual name of the repo. I haven't managed to make it work with `select()`. But this seems to be an issue with the suite's compilation of jq. It works with the official binaries, for some reason.

**Comment by 1480c1** [comment]
Hmm, I have also been trying to see if I could use the name instead of a hard coded index, but all of the commands lead to `not defined at <top-level>`

**Comment by 1480c1** [comment]
But it seems to work with the pacman installed one

**Comment by 1480c1** [comment]
could this be necessary? https://github.com/msys2/MINGW-packages/blob/854b9fc6cf46206a5bb74095905ef0f4cba388a4/mingw-w64-jq/no-undefined.patch

**Comment by wiiaboo** [comment]
That just fixes compiling issues, not runtime, I think.

**Comment by 1480c1** [comment]
I see. Hmmmmmmmmm

**Comment by wiiaboo** [comment]
Ok, I've tried using `--with-oniguruma=builtin` and it seemed to fix the issue with select. I guess mingw's oniguruma is broken.

**Comment by 1480c1** [comment]
Right now also compiling with built-in, but if that is so, then I will change it to use a built in oniguruma according to their readme

**Comment by 1480c1** [comment]
Doesn't work for me it seems

**Comment by 1480c1** [comment]
Very tempted to just use powershell

**Comment by 1480c1** [comment]
`($(Invoke-WebRequest https://raw.githubusercontent.com/KhronosGroup/Vulkan-ValidationLayers/master/scripts/known_good.json | ConvertFrom-Json).repos | Where-Object {$_.name -eq "Vulkan-Headers"}).commit`

**Comment by wiiaboo** [comment]
> Right now also compiling with built-in, but if that is so, then I will change it to use a built in oniguruma according to their readme

Yeah, but you need to either fix compiling it with the separate_conf or switch to compiling in-root.
Or compiling it before jq as a dependency. jq is overall more useful if we just use built-in or our own oniguruma instead of mingw's.

**Comment by 1480c1** [comment]
I'll try to see if we can build oniguruma and then jq instead of using built in oniguruma or pacman

**Comment by 1480c1** [comment]
I can confirm that it works if building oniguruma, jq, and vulkan-* from scratch

**Comment by 1480c1** [comment]
I will squash tomorrow morning after the win10 vm finishes. (win8 finished earlier due to the rest already being compiled)

File: build/media-suite_compile.sh [hunk 3]
```diff
@@ -1825,13 +1833,20 @@ if [[ $mpv != "n" ]] && pc_exists libavcodec libavformat libswscale libavfilter;
...
+        # Get the known_good json and get the commit version of the Vulkan-Headers.
+        # It is assumed that the version number has quotes around them (sed)
+        [[ "$jq" != y ]] && do_pacman_install jq
+        _Vulkan_Version=$(
+            ${curl_opts[@]} https://raw.githubusercontent.com/KhronosGroup/Vulkan-ValidationLayers/master/scripts/known_good.json |
+            jq -r '.repos[] | select(.name == "Vulkan-Headers").commit')
+        [[ "$jq" != y ]] && do_pacman_remove jq
         _check=(vulkan/vulkan.h)
-        if do_vcs "https://github.com/KhronosGroup/Vulkan-Headers.git#tag=v1.1.102" vulkan-headers; then
+        if do_vcs "https://github.com/KhronosGroup/Vulkan-Headers.git#tag=${_Vulkan_Version}" vulkan-headers; then
```
Comment (1480c1) [inline build/media-suite_compile.sh]:
```suggestion
        if do_vcs "https://github.com/KhronosGroup/Vulkan-Headers.git#tag=${_Vulkan_Version:-v1.1.106}" vulkan-headers; then
```

**Comment by wiiaboo** [comment]
The oniguruma adding could be separated into a commit by itself for easier reverting.
The "Add default version" commit could also be squashed into the Vulkan-Headers commit, since it's a change to code added there that isn't merged yet.

**Comment by 1480c1** [comment]
Sure

**Comment by wiiaboo** [comment]
Cool, thanks.

**Pull request merged**